Create an `Action` ButtonStyle

JPKribs opened the pull request:
### Summary

Resolves: https://github.com/jellyfin/Swiftfin/issues/1667

Originally Mentioned Here: https://github.com/jellyfin/Swiftfin/pull/1543#issuecomment-2896497162

I wanted to split this out into it's own PR / Discussion for better visibility.

---

This PR changes the action buttons from their current floating icons to instead use more a square button similar to tvOS. I try to preserve the Icon Color so it doesn't flip when the button goes from selected to unselected. To do this, I am using the same overlay color for the `PlayButton` so it's consistent. Un-selected buttons are `.gray` to account for either black or white icon colors.

All buttons grow to fill their container, identical to tvOS, except Compact Poster they will always be exact squares similar to the existing layout. 

---

Totally open to suggestions for this! I'm not positive this is what was meant in the [original comment](https://github.com/jellyfin/Swiftfin/pull/1543#issuecomment-2896497162) but this is where my mind went since this would bring iOS and tvOS in line with similar layouts.

---

### Screenshots

<details>

<summary>Cinematic</summary>

![Cinematic](https://github.com/user-attachments/assets/eb643406-56ae-4082-a436-9dd9fc1f0cf4)

</details>

<details>

<summary>Compact Poster</summary>

![Compact Poster](https://github.com/user-attachments/assets/acc2ea8b-9b07-4d85-928f-ebbe99183bab)

</details>

<details>

<summary>Generic / Simple Scroll View</summary>

![Episode](https://github.com/user-attachments/assets/0994eea1-056c-479f-9d90-5ad8ff9b72eb)

</details>

<details>

<summary>Collections</summary>

![Collection](https://github.com/user-attachments/assets/daa7a6a0-f679-466a-9fba-c67e34daa0bd)

</details>

<details>

<summary>iPadOS - Standard</summary>

![iPad Cinematic](https://github.com/user-attachments/assets/19b24731-0439-4473-961f-ac5eeb674c40)

</details>

<details>

<summary>iPadOS - Collections</summary>

![iPad Collections](https://github.com/user-attachments/assets/1fc1ff00-1f05-4125-bc5a-7f07c1196749)

</details>

<details>

<summary>Minimum Items vs Maximum Items</summary>

![Max vs Min Count](https://github.com/user-attachments/assets/615c3e41-f1f8-44c6-849c-c017f3a1c717)

</details>

JPKribs commented:
> Well, it looks like Apple is moving toward pill shaped buttons in tvOS 26 😆

The only constant is change I suppose lol.

I really hope there are some revisions in iOS 26 before the final release. I personally felt like a lot of their demos were really difficult to read with the layered transparency

JPKribs commented:
I moved the `ActionButton` to a `ButtonStyle` on iOS. It looks identical to what I have in my screenshot but the implementation feels a lot cleaner for both Buttons and Menus.

I've tested this as well, I believe we should be able to use this if we want to make the `PlayButton` on iOS use this same styling. I haven't included that in this same PR but below are what this looks like using the `.buttonStyle(.action)` for the `PlayButton`:

```
            Button(title, systemImage: "play.fill") {
                if let playButtonItem = viewModel.playButtonItem, let selectedMediaSource = viewModel.selectedMediaSource {
                    mainRouter.route(to: \.videoPlayer, OnlineVideoPlayerManager(item: playButtonItem, mediaSource: selectedMediaSource))
                } else {
                    logger.error("No media source available")
                }
            }
            .buttonStyle(.action)
            .foregroundStyle(accentColor.overlayColor, accentColor)
            .font(.callout)
            .fontWeight(.semibold)
            .disabled(viewModel.playButtonItem == nil)
```

<details>

<summary>Compact</summary>

![Simulator Screenshot - iPhone 16 Pro - 2025-06-20 at 15 52 05](https://github.com/user-attachments/assets/4c0c9b7d-7738-42ed-96d1-ef4e2a76871c)

</details>

<details>

<summary>Wide</summary>

![Simulator Screenshot - iPhone 16 Pro - 2025-06-20 at 15 52 22](https://github.com/user-attachments/assets/6711b4ac-8bee-45bf-b30f-3cf770098b53)

</details>

---

I wish I could do the same in tvOS but I was finding any buttons you have to build EVERYTHING including the animations for pressing, focus, etc. So I left tvOS mostly alone but I tried to mirror the `.foregroundStyle` between them.

I am going to stop working on this now so this isn't a moving target to review but this should be done and ready for a second look.

LePips reviewed (changes requested):
Apologies for not keeping up with this PR. For these buttons I've always wanted a tinted material instead of a solid color and I've added those in my latest commit. Play around with that for these buttons to see what it looks like.

Edit: apologies for no example, usage would look somewhat like the following:

```swift
Button {} label: {
    Image(systemName: "heart.fill")
        .foregroundStyle(.white)
        .padding(10)
        .background(
            TintedMaterial()
        )
        .tint(.green)
        .cornerRadius(10)
}
```

JPKribs commented:
> Apologies for not keeping up with this PR. For these buttons I've always wanted a tinted material instead of a solid color and I've added those in my latest commit. Play around with that for these buttons to see what it looks like.
> 
> Edit: apologies for no example, usage would look somewhat like the following:
> 
> ```swift
> Button {} label: {
>     Image(systemName: "heart.fill")
>         .foregroundStyle(.white)
>         .padding(10)
>         .background(
>             TintedMaterial()
>         )
>         .tint(.green)
>         .cornerRadius(10)
> }
> ```

I think I'm off. I have this on my latest commit:

https://github.com/user-attachments/assets/458cd428-bb55-4321-8647-1b841f3a030b

https://github.com/user-attachments/assets/c481bee4-eeaf-48e7-a353-a216f552802f

**Simple**

<img width="505" height="532" alt="Screenshot 2025-08-07 at 13 49 29" src="https://github.com/user-attachments/assets/f8f7623d-f266-413d-9ba4-368d160f401c" />

**Poster**

<img width="231" height="167" alt="Screenshot 2025-08-07 at 13 49 57" src="https://github.com/user-attachments/assets/1d5dd8cb-b643-4954-8e95-9cb79c37d871" />

I added a 50% gray tint because otherwise I would get:

<img width="331" height="240" alt="Screenshot 2025-08-07 at 13 52 03" src="https://github.com/user-attachments/assets/d023744a-ce4f-4465-93eb-5a169b708e7e" />

<img width="1206" height="2622" alt="Simulator Screenshot - iPhone 16 Pro - 2025-08-07 at 13 52 10" src="https://github.com/user-attachments/assets/2dad7901-7c0b-4ef5-923e-366ba4490694" />

I don't love how this looks though. I'm not sure what our best route is for making the icon visible since the tinting makes it easy to lose it

JPKribs commented:
It's also coming in a little dark. I think, just because the material we were using is `.dark`. IMO, it might look better if we had the play button using the same style just so it's more cohesive. Especially if we keep the `IsPlayed` button as `accentColor` (which IMO I don't like them as the same color but 🤷🏻‍♂️)

JPKribs commented:
Cleaned this up a little. Made this more transparent to see the color through and made the text always white.

https://github.com/user-attachments/assets/f74a63e2-daca-40c9-9b87-8767aac56d56

https://github.com/user-attachments/assets/719cc6dc-4320-4ed5-9291-610031bb806f

https://github.com/user-attachments/assets/695cdbbe-49ab-44e4-86a1-4c555efefba4

https://github.com/user-attachments/assets/9eb27b3a-3a2d-4335-bd0c-05da0fa911eb

JPKribs commented:
Those were a big help, thank you! I have this in a place where I think this looks good. The last question is whether we'd want to fully mirror Apple where the text for the button is the accent color. I like what Apple does for Apple Music and Podcasts but they also have a static accent color so they've selected colors that are more distinct and less likely to be hard to see:

<img width="1308" height="744" alt="Screenshot 2025-08-10 at 13 29 38" src="https://github.com/user-attachments/assets/0a7ac8b4-ef82-4c32-8e72-a6cf8fd7c77f" />

In Swiftfin, this is what I have in my last commit. Please let me know if this is not the direction we want to go:

<img width="275" height="115" alt="Screenshot 2025-08-10 at 13 19 30" src="https://github.com/user-attachments/assets/172405a0-cd66-4d5a-aa24-0652c62616f6" />

<img width="276" height="134" alt="Screenshot 2025-08-10 at 13 19 57" src="https://github.com/user-attachments/assets/89ca3a78-5246-4433-ac0e-4a69066e26d2" />

<img width="276" height="132" alt="Screenshot 2025-08-10 at 13 20 27" src="https://github.com/user-attachments/assets/1e9675c2-241a-4242-b325-32ef4b99611b" />

<details>

<summary>How Collections Look</summary>

<img width="1206" height="2622" alt="Simulator Screenshot - iPhone 16 Pro - 2025-08-10 at 13 16 36" src="https://github.com/user-attachments/assets/187fe8fa-5dcb-4a68-b7b6-6927645c777f" />

</details>

LePips commented:
> the text for the button is the accent color.

No, just please have the styling as it was prior.

JPKribs commented on Swiftfin/Views/ItemView/Components/ActionButtonHStack/ActionButtonHStack.swift:
```diff
@@ -52,80 +50,75 @@ extension ItemView {
                         viewModel.send(.toggleIsPlayed)
                     }
+                    .buttonStyle(.tintedMaterial(tint: .jellyfinPurple))
+                    .foregroundStyle(.white)
```
Testing this on Light Mode for Collections is hard to see. I originally used .primary so SimpleScrollView could use .black for light mode.
